test/test_metapackage.py

```diff
@@ -76,6 +86,16 @@ def test_validate_metapackage(self):
                 else:
                     _validate_metapackage(path, package)
 
+    def test_collect_warnings(self):
+        """Tests warnings collection"""
+        warnings = []
+        pkgs_dict = find_packages(test_data_dir, warnings=warnings)
+
+        warnings.sort()
+        test_expected_warnings.sort()
+        for w, we in zip(warnings, test_expected_warnings):
+            assert w == we, 'Warning message "%s" != "%s"' % (w,we)
```

Comment:
Asserting on the identity of the items will result in a not very helpful error message. Instead please just assert that both sorted lists are equal.